src/datagrid/DataGrid.tsx

```diff
@@ -267,12 +267,12 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
     };
 
     // Function to update datagrid rows
-    updateRows = (rows: DataGridRow[], totalItems: number) => {
+    updateRows = (rows: DataGridRow[], totalItems?: number) => {
         const updatedRows = this.updateRowIDs(rows);
         let {pagination} = this.state;
 
         // update pagination footer
-        if (pagination) {
+        if (pagination && totalItems) {
```

Comment:
In which case the totalItems will be different from rows.length()?

Reply:
@psivakumaran I have mention that in PR description. For pagination total-items are different than length of current rows.